### DISCUSSION TITLE
Using timeseries grouping inside the timeseries index

### DISCUSSION THREAD

**Discussion by Azizollahi** [ravendb/ravendb#12641] [Q&A] [answered]
I have read [this](https://ravendb.net/docs/article-page/5.2/csharp/document-extensions/timeseries/indexing) and [this](https://ravendb.net/docs/article-page/5.2/csharp/document-extensions/timeseries/querying/aggregation-and-projections)

I have a collection with a series called "TradePrice" I want to create a [timeseries index](https://ravendb.net/docs/article-page/5.0/csharp/document-extensions/timeseries/indexing#abstracttimeseriesindexcreationtask) then by [map-reduce](https://ravendb.net/docs/article-page/5.0/csharp/document-extensions/timeseries/indexing#map-reduce-time-series-index) 
 and [timeseries grouping](https://ravendb.net/docs/article-page/5.2/csharp/document-extensions/timeseries/querying/aggregation-and-projections#query-examples) in that index create a new collection that has pre-generated 1-minute data like below for future use:
{
"Time"= "09:00",
"Open"= 125,
"Close"= 22,
"High"= 255,
"Low"= 10
},
{
"Time"= "09:01",
"Open"= 52,
"Close"= 45,
"High"= 100,
"Low"= 20
}

Is this possible?

**Comment by aviv86** [comment, marked as answer]
Hi
Something like this, maybe ?
```
private class TimeSeries_Index : AbstractTimeSeriesIndexCreationTask<Entity, TimeSeries_Index.Result>
{
	public class Result
	{
		public DateTime Time { get; set; }

		public double Open { get; set; }

		public double Close { get; set; }

		public double High { get; set; }

		public double Low { get; set; }

	}

	public TimeSeries_Index()
	{
		AddMap("TradePrice", segments =>
			from segment in segments
			from entry in segment.Entries
			select new
			{
				Time = entry.Timestamp,
				Open = entry.Values[0],
				Close = entry.Values[1],
				High = entry.Values[2],
				Low = entry.Values[3],
			});
		Reduce = results =>
			from r in results
			group r by new DateTime(r.Time.Year, r.Time.Month, r.Time.Day, r.Time.Hour, r.Time.Minute, 0) into g
			select new
			{
				Time = g.Key,
				Open = g.Average(x => x.Open),
				Close = g.Average(x => x.Close),
				High = g.Max(x => x.High),
				Low = g.Min(x => x.Low)
			};

		OutputReduceToCollection = "TradePriceBy1Minute";
	}
}
```

**Reply by Azizollahi** [reply]
Yes, we can do it that way but now why should I save my data as TimeSeries? because now I can save them as a normal collection and then do the same thing with normal indexes.

**Reply by ayende** [reply]
That wouldn't make any sense, and indeed, there isn't a point in doing so.
What you can do is store the data as time series and define a rollup if needed.

Although, to be honest, you should just query on the raw details unless you have a *lot*

**Comment by kamranayub** [comment]
I'm currently working on a sample demo app and database to showcase time series and market data, like this. I'll be sure to post here once it's available!

![image](https://user-images.githubusercontent.com/563819/129582812-29ddf00d-65e1-424d-bddd-55595391087d.png)
